pkg/controller/servicebinding/secret.go

```diff
@@ -10,98 +12,108 @@ import (
 	"k8s.io/client-go/dynamic"
 
 	"github.com/redhat-developer/service-binding-operator/pkg/converter"
-	"github.com/redhat-developer/service-binding-operator/pkg/log"
 )
 
-// secret represents the data collected by this operator, and later handled as a secret.
-type secret struct {
-	logger *log.Log          // logger instance
-	client dynamic.Interface // Kubernetes API client
-	ns     string
-	name   string
-}
-
 // buildResourceClient creates a resource client to handle corev1/secret resource.
-func (s *secret) buildResourceClient() dynamic.ResourceInterface {
+func buildResourceClient(client dynamic.Interface, namespace string) dynamic.ResourceInterface {
```

Comment:
Before this symbol was scoped by the `secret` type, but now it doesn't and the function (which is available to the entire `servicebinding` package) has a very opaque name. Please change it to `buildSecretClient(client, namespace)` or something more meaninful instead.